### ISSUE TITLE
fraud: FrauSync shoulduse existing connections

### ISSUE THREAD

**Issue by Wondertan** [celestiaorg/celestia-node#1236]
When fraud.Service starts it starts requesting new peers for potential frauds. However, this excludes connections to peers made before fraud.Service started.

**Comment by renaynay** [comment]
The reason we didn't use existing connections before is because it was assumed those existing connections were "bootstrappers" so the protocol wanted to diversify the peers from which it requests fraud, correct? 

I think we should maintain this kind of design though -- we can request 1 bootstrapper + 4 other peers or so, but IMO should still ensure that the majority of peers it requests frauds from are not bootstrappers.

**Comment by Wondertan** [comment]
Yeah, and the solution to the mentioned problem is easy. After making [`fraudRequests`](https://github.com/celestiaorg/celestia-node/blob/main/fraud/service.go#L25) configurable, we just need to ensure its amount is always +n where n is the number of bootstrappers.

**Comment by Wondertan** [comment]
Marking this issue as a good first one. It is slightly harder than other good first issues, as it requires looking into libp2p, but it should be fun. Anyone looking into it, pls don't hesitate to ask further questions.

**Comment by Jpserrat** [comment]
Hey, @Wondertan 
May I work on this? I know I have been giving you guys some trouble, but I promise to do better this time haha

**Comment by Jpserrat** [comment]
@Wondertan If I got it right, we are now getting new peer connections from the BusEvent and not using the ones we already have at the peer store. Now we should also iterate over the peer store ones.
From what you guys were discussing, should I use a static number of fraud requests and based on this static number change the request made to new connections or should the number of fraud requests to new connections change based on the number of requests made to bootstrappers?
Another question that I have, may I assume that all new connections are not bootstrappers?

**Comment by Wondertan** [comment]
@Jpserrat, I am currently OOO office so asking @vgonkivs to help as an original implementer of the fraud package

**Comment by Jpserrat** [comment]
Thanks!
Hey @vgonkivs, can you give me some help on this?

**Comment by vgonkivs** [comment]
Hey @Jpserrat, will get back to you once all urgent tasks will be finished(probably tomorrow).

**Comment by renaynay** [comment]
@Jpserrat you want to work on this still?

**Comment by Jpserrat** [comment]
Hey @renaynay, sure!

**Comment by vgonkivs** [comment]
Hello, @Jpserrat. Sorry, for sooo long response. I will help with this issue. Answering in your questions:

> If I got it right, we are now getting new peer connections from the BusEvent and not using the ones we already have at the peer store. Now we should also iterate over the peer store ones.

Correct. FraudService keeps a host, that has a peerStore inside.

>  should I use a static number of fraud requests and based on this static number change the request made to new connections or should the number of fraud requests to new connections change based on the number of requests made to bootstrappers?

You can add one more constant `boostrapperRequest`(I think 3 will be enough) and request them before subscription to the EventBus.

> may I assume that all new connections are not bootstrappers?
Yes

**Comment by Jpserrat** [comment]
Thanks @vgonkivs!
